### DISCUSSION TITLE
What is the best bar for i3?

### DISCUSSION THREAD

**Discussion by adriandelgg** [i3/i3#5683] [Q&A]
Hi everyone,

I'm currently using Polybar, but I'm wondering if there's a better alternative anywhere.

I'm not a fan of the default i3bar, but I've heard of alternatives like i3status, i3blocks, and bumblebees status.

Looking for something customizable, simple, bug-free as possible, integrates well with i3, and looks nice.

**Comment by okraits** [comment]
i3status, i3blocks and bumblebee status are not bar applications but statusline generators.

i3bar with i3blocks is customizable, stable and there's nothing better integrated.

**Reply by adriandelgg** [reply]
Thanks for responding. Why would someone pick `i3blocks` over `polybar`? It seems that `polybar` is better than `i3blocks`.

**Reply by okraits** [reply]
i3blocks is easy to configure and is easily extendable with scripts. It is mature and light on resources.

Why do you think polybar is better, and in which regard? Apart from that, they serve a different purpose.

**Reply by adriandelgg** [reply]
I configured i3blocks, but I don't like how you can't put the blocks in the center of the bar. 

I just see polybar everywhere like on r/unixporn and stuff, and it looks like it's more customizable.

**Reply by orestisfl** [reply]
It's important to note the difference between i3blocks and i3bar. i3blocks just outputs text that i3bar interprets to display information on the bar. The limitation of not centering blocks is in i3bar, not i3blocks.

> Looking for something customizable, simple, bug-free as possible, integrates well with i3, and looks nice.

It does seem that polybar satisfies these criteria for you, is there something you don't like?

**Reply by andrei-korshikov** [reply]
Also see https://www.reddit.com/r/i3wm/wiki/faq/i3i3bari3status/

If you are asking about the **bar**, then we are talking about i3bar vs Polybar.

"The best" means "suits your specific needs", so… just try them all. Really, that's just inevitable:D

> Why would someone pick i3blocks over polybar? 

As for me, Polybar is an overkill — I don't need all these _unix-porny_ things, moreover they distract me. Plus another configuration language... The only thing I have in my status line is the audio volume (yes, no time, no load, no free space, no network status, etc.) Of course, I would not ever think to put the audio volume to the center of the bar:) So, based on my workflow and my perception, i3bar + i3blocks is better (moreover, i3bar + i3status would suit me just fine), but, as you see, that's very subjective.

Edit: totally forgot to mention [i3status-rust](https://github.com/greshake/i3status-rust) (or, in the scope of our discussion, `i3bar`+`i3status-rust`;) ). Looks [nice](https://github.com/greshake/i3status-rust/blob/master/doc/themes.md#available-themes) (of course, I use the default "plain" theme because it's the most unobtrusive:D), and pretty configurable. E.g. for my "audio volume only" purpose, I can set different colors for muted microphone vs muted speaker, mouse click events (mute, start "pavucontrol --tab=XXX"), hide "microphone" block if "Analog Stereo Output" profile is used, inverse mouse wheel… And, last but not least, I can finally rid of that signals business: when the volume is changed by some external program, it is "just" updated in the status, without things like ` killall -SIGUSR1 <your_statusline_generator>` or refreshing on timeout. So, yes, I do recommend)
